### DISCUSSION TITLE
Micro-frontend / Module federation with Quasar Apps

### DISCUSSION THREAD

**Discussion by vinicius73** [quasarframework/quasar#11115] [Ideas / Proposals]
Hi everyone.
I will move a old project (Vue.js v2 + Vuetify) to Quasar. I am used to use quasar in other projects and I enjoy a lot.
That project has a lot of modules/sections which will works better with a micro-frontends approach.

Unfortunately I don't find any info about quasar + micro-frontends. I know quasar CLI works over webpack, and I can use module federation, but before I start I want to know from you if had you any knowledge about it and what it the better what to do it with Quasar.

My guess is I will not able to use Quasar CLI in non shell build, it looks sad, but I don't know.

**Comment by schupp13** [comment]
Curious what route you ended up taking to get module federation to work. Did you end up making a vue-cli project, then bring in quasar into that project? Or did you figure out how to do it in a quasar-cli project? I've also grown to love quasar and want to use it to make a suite of SPAs and share components between them? I've been trying for a long time to get it working on a quasar-cli project....but keep bouncing around the same errors. Downfall of not having access to the main.js file I guess ( to make a bootloader.js ) . If you have any advice or things to look out for, I would greatly appreciate it.

**Reply by vinicius73** [reply]
I give up from the original idea of use micro front-ends @schupp13 

Switching to Quasar + Vite solves the main problem: development performance/experience.

As vite loads all code in lazy mode (if you use lazy imports), we do not have our original problem anymore.
